crates/ark/src/lsp/documents.rs

```diff
@@ -224,4 +225,11 @@ mod tests {
         let point = compute_point(Point::new(0, 0), "abcdefghi\n");
         assert_eq!(point, Point::new(1, 0));
     }
+
+    #[test]
+    fn test_document_starts_at_0_0_with_leading_whitespace() {
+        let document = Document::new("\n\n# hi there", None);
+        let root = document.ast.root_node();
+        assert_eq!(root.start_position(), Point::new(0, 0));
+    }
```

Comment:
This is the test that ensures that the `program` node should start at `(0, 0)` and cover the whole document.

Reply:
It was previously possible for the program node to start at, say, `(3, 0)` if thats where the first content of the document is. Then we'd request something like "find closest node to" `(0, 1)` and that in theory should return the `program` node as the top level node which covers the whole document, but since it started at `(3, 0)` we'd get a panic (or at least an error log).

This isn't every source of these errors, but it was one I saw a decent bit